scripts/resolve_deps/main.go

```diff
@@ -0,0 +1,212 @@
+		os.Exit(0)
+	}
+	fmt.Printf("Plugin and Gloo Enterprise dependencies do not match after %d merge attempts!\n", mergeAttempt)
+
+	// 1. Write the report to stdout
+	reportBytes, err := json.MarshalIndent(nonMatchingDeps, "", "  ")
+	if err != nil {
+		fmt.Printf("Failed to marshall error report to JSON: %s/n", err.Error())
+		os.Exit(1)
+	}
+	fmt.Println(string(reportBytes))
+
+	// 2. Write the report to a file
+	fmt.Printf("Writing error report file [%s] after %d merge attempts\n", errorReportFile, mergeAttempt)
+	if err := ioutil.WriteFile(errorReportFile, reportBytes, 0644); err != nil {
+		fmt.Printf("Failed to write error report file: %s/n", err.Error())
+	}
+
+	// 3. Create a file with suggested changes to go.mod
+	if err := createSuggestionsFile(nonMatchingDeps); err != nil {
+		fmt.Printf("Failed to create suggestions file: %s/n", err.Error())
+	}
+	os.Exit(1)
+}
+
+func resolveDependencies(moduleFilePath, glooDependenciesFilePath string, mergeAttempt int) ([]checks.DependencyInfoPair, error) {
+	var (
+		nonMatchingDeps []checks.DependencyInfoPair
+		moduleInfo      *checks.ModuleInfo
+		err             error
+	)
+	suggestionModuleFileName := moduleFilePath
+	for i := 1; mergeAttempt > 0 && i <= mergeAttempt; i++ {
+		if moduleInfo, nonMatchingDeps, err = checks.CompareModuleFile(suggestionModuleFileName, glooDependenciesFilePath); err != nil {
+			return nil, errors.Wrapf(err, "failed to compare dependencies")
+		}
+
+		if len(nonMatchingDeps) == 0 {
+			if suggestionModuleFileName != moduleFilePath {
+				if err = os.Rename(suggestionModuleFileName, moduleFilePath); err != nil {
+					return nil, errors.Wrapf(err, "failed to rename temp suggestions module '%s' to current '%s' file\n", suggestionModuleFileName, moduleFilePath)
+				}
+			}
+
+			return nonMatchingDeps, nil
+		}
+		fmt.Println("Plugin and Gloo Enterprise dependencies do not match!")
+		if i < mergeAttempt {
+			fmt.Printf("Merging dependencies and start comparing again (attempt: %d)\n", i)
+		}
+
+		mergeDependencies(moduleInfo.Dependencies, nonMatchingDeps)
+
+		suggestionFilesDir := filepath.Dir((filepath.Join(tempDirName, moduleFilePath)))
+		if err := os.MkdirAll(suggestionFilesDir, os.ModePerm); err != nil {
+			return nil, errors.Wrapf(err, "failed to create temp suggestions dir '%s' file\n", suggestionFilesDir)
+		}
+
+		suggestionModuleFileName = fmt.Sprintf("%s/%s-%d", suggestionFilesDir, moduleFilePath, i)
+		if err = createPluginModuleFile(suggestionModuleFileName, moduleInfo); err != nil {
+			return nil, errors.Wrapf(err, "failed to write new merged '%s' file\n", moduleFilePath)
+		}
+	}
+	return nonMatchingDeps, err
+}
+
+func mergeDependencies(pluginDependencies map[string]checks.DependencyInfo, nonMatchingDeps []checks.DependencyInfoPair) {
+	for _, dep := range nonMatchingDeps {
+		pluginDependencies[dep.Plugin.Name] = dep.Gloo
+	}
+}
+
+func createSuggestionsFile(nonMatchingDeps []checks.DependencyInfoPair) error {
```

Comment:
Not sure we really need this anymore. In general, this PR appears to have some copy-pasted code from the other script, so it'd be good to remove that script as part of this PR too.